tests/themis/themis_seccure_message.c

```diff
@@ -215,14 +364,197 @@ static void themis_secure_message_test(){
                     "Hit http://ftp.us.debian.org[11] wheezy/contrib Translation-en";
 
   size_t message_length=strlen(message);
-  
-  testsuite_fail_if(themis_secure_signed_message_generic_test(RSA_ALG, message, message_length), "themis secure signed message (RSA)");
-  testsuite_fail_if(themis_secure_signed_message_generic_test(EC_ALG, message,message_length), "themis secure signed message (EC)");
-  testsuite_fail_if(themis_secure_encrypted_message_generic_test(RSA_ALG, message, message_length), "themis secure encrypted message (RSA)");
-  testsuite_fail_if(themis_secure_encrypted_message_generic_test(EC_ALG, message,message_length), "themis secure encrypted message (EC)");
+
+  testsuite_fail_if(generic_themis_secure_message_encrypt_decrypt_test(RSA_ALG, message, message_length), "themis secure encrypt/decrypt message (RSA)");
+  testsuite_fail_if(generic_themis_secure_message_encrypt_decrypt_test(EC_ALG, message, message_length), "themis secure encrypt/decrypt message (EC)");
+  testsuite_fail_if(generic_themis_secure_message_sign_verify_test(RSA_ALG, message, message_length), "themis secure sign/verify message (RSA)");
+  testsuite_fail_if(generic_themis_secure_message_sign_verify_test(EC_ALG, message, message_length), "themis secure sign/verify message (EC)");
+
+  testsuite_fail_if(themis_secure_signed_message_generic_test(RSA_ALG, message, message_length), "(deprecated) themis secure signed message (RSA)");
+  testsuite_fail_if(themis_secure_signed_message_generic_test(EC_ALG, message,message_length), "(deprecated) themis secure signed message (EC)");
+  testsuite_fail_if(themis_secure_encrypted_message_generic_test(RSA_ALG, message, message_length), "(deprecated) themis secure encrypted message (RSA)");
+  testsuite_fail_if(themis_secure_encrypted_message_generic_test(EC_ALG, message,message_length), "(deprecated) themis secure encrypted message (EC)");
+}
+
+static void themis_secure_message_encrypt_decrypt_api_test(void){
+  themis_status_t status=THEMIS_FAIL;
+
+  uint8_t private_key[MAX_KEY_SIZE]={0};
+  uint8_t public_key[MAX_KEY_SIZE]={0};
+  size_t private_key_length = sizeof(private_key);
+  size_t public_key_length = sizeof(public_key);
+
+  uint8_t plaintext[MAX_MESSAGE_SIZE]={0};
+  uint8_t encrypted[MAX_MESSAGE_SIZE+1024]={0};
```

Comment:
why +1024?

Reply:
That's an arbitrary constant which should be enough for whatever authentication data Secure Message appends to the encrypted message (the tests will fail if it's not enough). i'll name it.